pil-analyzer/src/statement_processor.rs

```diff
@@ -463,26 +459,28 @@ where
         Ok(n.into())
     }
 
-    fn evaluate_expression_to_int(&self, expr: parsed::Expression<T>) -> Result<BigInt, EvalError> {
-        evaluator::evaluate_expression(
+    fn evaluate_expression_to_int(&self, expr: parsed::Expression) -> Result<BigInt, EvalError> {
+        // TODO we should maybe implement a separate evaluator that is able to run before type checking
+        // and is field-independent (only uses integers)?
```

Comment:
Yes, but I think array sizes should not depend on the field. It might be tempting to do that (looking at poseidon), but I think it would be better to then just use a different machine instead of the same machine with differently-sized arrays.